pkg/profilecreator/profilecreator_test.go

```diff
@@ -54,6 +54,38 @@ var _ = Describe("PerformanceProfileCreator: MCP and Node Matching", func() {
 			Expect(matchedNodes[0].GetName()).To(Equal("dhcp19-232-239.fci1.kni.lab.eng.bos.redhat.com"))
 		})
 	})
+
+	Context("Ensure the correct MCP selector is used", func() {
+		It("should detect the cnf-worker MCP selector", func() {
+			mcp, err := GetMCP(mustGatherDirPath, "worker-cnf")
+			Expect(err).ToNot(HaveOccurred())
+
+			mcpSelector, err := GetMCPSelector(mcp, mcps)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(mcpSelector)).To(Equal(1))
+
+			for key, value := range mcpSelector {
+				Expect(key).To(Equal("machineconfiguration.openshift.io/role"))
+				Expect(value).To(Equal("worker-cnf"))
+				break
+			}
+		})
+
+		It("should detect the worker MCP selector", func() {
+			mcp, err := GetMCP(mustGatherDirPath, "worker")
+			Expect(err).ToNot(HaveOccurred())
+
+			mcpSelector, err := GetMCPSelector(mcp, mcps)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(mcpSelector)).To(Equal(1))
+
+			for key, value := range mcpSelector {
+				Expect(key).To(Equal("pools.operator.machineconfiguration.openshift.io/worker"))
+				Expect(value).To(Equal(""))
+				break
+			}
+		})
+	})
```

Comment:
do we want to add negative tests -  e.g. tests that check the error path, not just the happy path? perhaps in a later PR?

Reply:
I am all for adding more tests if it helps. I wonder how a negative test would look like and what would be the added value.  A non existing MCP is already tested before the MCP selector, so not an issue. An interesting "fail" path would be a cluster configuration where we cannot find an unique MCP selector for the target MCP. But I don't have a concrete example... maybe we should add it to the E2E tests if we can come up with something.